tests/catch/unit/device/hipSetGetDevice.cc

```diff
@@ -83,6 +87,46 @@ TEST_CASE("Unit_hipGetSetDevice_MultiThreaded") {
   HIP_CHECK_THREAD_FINALIZE();
 }
 
+TEST_CASE("Unit_hipSetGetDevice_Positive_Threaded_Basic") {
+  class HipSetGetDeviceThreadedTest : public ThreadedZigZagTest<HipSetGetDeviceThreadedTest> {
+   public:
+    void TestPart1() { HIP_CHECK(hipSetDevice(0)); }
+
+    void TestPart2() {
+      HIP_CHECK_THREAD(hipSetDevice(1));
+      HIP_CHECK_THREAD(hipMalloc(&ptr, 2 * 1024 * 1024));
+    }
+
+    void TestPart3() {
+      int device = -1;
+      HIP_CHECK_THREAD(hipGetDevice(&device));
+      REQUIRE_THREAD(device == 0);
+      device = -1;
+      // To check if set device worked properly, outside of hipGetDevice
+      HIP_CHECK_THREAD(hipPointerGetAttribute(&device, HIP_POINTER_ATTRIBUTE_DEVICE_ORDINAL,
+                                              reinterpret_cast<hipDeviceptr_t>(ptr)));
+      REQUIRE_THREAD(device == 1);
+    }
+
+    void TestPart4() {
```

Comment:
Remove the empty lines between TestPart1/2/3/4

Reply:
The suggested change is implemented. Please proceed with the review.